storage/mysql/provider.go

```diff
@@ -107,3 +120,32 @@ func (s *mysqlProvider) AdminStorage() storage.AdminStorage {
 func (s *mysqlProvider) Close() error {
 	return s.db.Close()
 }
+
+func registerTLSConfig() error {
+	if *mySQLTLSCA == "" {
+		return nil
+	}
+
+	rootCertPool := x509.NewCertPool()
+	pem, err := os.ReadFile(*mySQLTLSCA)
+	if err != nil {
+		return err
+	}
+	if ok := rootCertPool.AppendCertsFromPEM(pem); !ok {
+		return errors.New("failed to append PEM")
+	}
+	serverName := ""
+	if *mySQLServerName != "" {
+		serverName = *mySQLServerName
+	} else {
+		serverName = os.Getenv("MYSQL_HOSTNAME")
+	}
+	err = mysql.RegisterTLSConfig("custom", &tls.Config{
+		RootCAs:    rootCertPool,
+		ServerName: serverName,
+	})
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
Since you're not wrapping the error, this can just be shortened to `return mysql.RegisterTLS...`